lib/setupmanagers/qemuhck/dhcpd.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+
+# AutoHCK module
+module AutoHCK
+  # Dhcpd class
+  class Dhcpd
+    def initialize(project)
+      @ip_segment = project.config['ip_segment']
+      @workspace_path = project.workspace_path
+    end
+
+    def run(dump_only: false)
+      conf_path = File.join(@workspace_path, 'dhcpd.conf')
+      leases_path = File.join(@workspace_path, 'dhcpd.leases')
+
+      dump_conf conf_path
+      FileUtils.touch leases_path
+
+      if dump_only
+        IO.write File.join(@workspace_path, 'dhcpd_manual.sh'), <<~SH
+          #!/usr/bin/env bash
+          dhcpd -f -d -cf #{conf_path} -lf #{leases_path} --no-pid
+        SH
+      else
+        @pid = spawn('dhcpd', '-f', '-d', '-cf', conf_path, '-lf', leases_path, '--no-pid')
+      end
+    end
+
+    def abort
+      return if @pid.nil?
+
+      Process.kill(:KILL, @pid)
+      Process.wait(@pid)
+    end
+
+    private
+
+    # rubocop:disable Metrics/MethodLength
+    def dump_conf(conf_path)
+      File.open(conf_path, 'w') do |f|
+        f.write <<~CONF
+          default-lease-time 600;
+          max-lease-time 7200;
+
+          ddns-update-style none;
+
+          subnet #{@ip_segment}0 netmask 255.255.255.0 {
+
+          option routers #{@ip_segment}1;
+          range #{@ip_segment}200 #{@ip_segment}254;
+        CONF
+
+        (2..99).each do |i|
+          i2 = format('%02d', i)
+
+          f.write <<~CONF
+            host qemuhck#{i2} {
+              hardware ethernet 56:00:#{i2}:00:dd:dd;
+              fixed-address #{@ip_segment}#{i};
+            }
+
+            host virthck#{i2} {
+              hardware ethernet 56:00:#{i2}:00:#{i2}:dd;
+              fixed-address #{@ip_segment}#{i};
+            }
```

Comment:
AutoHCK does not have virthck support now. This is present in the DHCP configuration because a user can reuse the same bridge for VirtHCK. So just remove this part.